utils/config.py

```diff
@@ -0,0 +1,66 @@
+from adafruit_hid.keycode import Keycode
+
+from utils.constants import OS_WINDOWS
+from utils.serial_tools import get_serial_data, trigger_and_get
+
+
+class Config:
+    KEYCODE_NUMBER_MAPPING = {
+        0: Keycode.ZERO,
+        1: Keycode.ONE,
+        2: Keycode.TWO,
+        3: Keycode.THREE,
+        4: Keycode.FOUR,
+        5: Keycode.FIVE,
+        6: Keycode.SIX,
+        7: Keycode.SEVEN,
+        8: Keycode.EIGHT,
+        9: Keycode.NINE,
+    }
+    times = 0
+
+    def __init__(self) -> None:
+        self.OS_SETTING = OS_WINDOWS
+        self.PREVIOUS_APP_SETTING = []
+
+        self.pinned_apps = {
+            "chrome": "1",
+            "files": "2",
+            "phpstorm": "3",
+            "intellij": "4",
+            "vscode": "5",
+            "postman": "6",
+            "slack": "7",
+            "terminal": "8",
+        }
+        self.email = "mtyemail@email.com"
+        self.host = {}
+
+        self.reload_config(False)
+
+    def reload_config(self, check_only=True):
+        try:
+            if check_only:
+                if data := get_serial_data():
+                    print(data)
+                    self.host = data
+                    return True
+            else:
+                self.host = trigger_and_get()
+                # if timestamp := self.host.get('timestamp'):
```

Comment:
I have to take another look at this, but the idea here is to allow the MacroPad to be aware of the current time. This would allow for time based action (such as playing a reminder tune or similar).

Reply:
Yeah, I like that idea. Maybe there's a way to set the time from the host when the macropad is connected and then keep it ticking on the device after the fact.

But, since this PR is pretty large, can we save that for a future PR?

Reply:
Really, with the exception of app switching, possible use of the data passing feature should come after this PR. 

You'll be delighted to know that what you are describing as far as the device managing the time itself is exactly how it works 😄 

Strictly speaking, updating the time every time data is passed would not be needed, but it would address issues with time drifting.